projects/igniteui-angular/src/lib/calendar/calendar-base.ts

```diff
@@ -656,9 +654,15 @@ export class IgxCalendarBaseDirective implements ControlValueAccessor {
         protected keyboardNavigation?: KeyboardNavigationService,
         protected cdr?: ChangeDetectorRef,
     ) {
-        this.locale = _localeId;
+        initi18n(_localeId);
+        this._defaultLocale = getCurrentI18n();
         this.viewDate = this.viewDate ? this.viewDate : new Date();
         this.initFormatters();
+
+        getI18nManager().onResourceChange((args: ResourceChangeEventArgs) => {
+            this._defaultLocale = args.newLocale;
```

Comment:
This works for most parts of the calendar, however there's a `formattedMonth` func that currently does not take the locale in consideration, so the month in the calendar remains in the default locale:

<img width="1368" height="484" alt="image" src="https://github.com/user-attachments/assets/317363ad-70e4-4c1b-adcc-6142c5d9ba5c" />